src/shared/treeview/awsTreeErrorHandlerNode.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { TreeItemCollapsibleState } from 'vscode'
+import { ErrorNode } from '../../lambda/explorer/errorNode'
+import { AWSTreeNodeBase } from './awsTreeNodeBase'
+
+export abstract class AWSTreeErrorHandlerNode extends AWSTreeNodeBase {
+    protected errorNode?: ErrorNode
+
+    protected constructor(
+        label: string,
+        collapsibleState?: TreeItemCollapsibleState
+    ) {
+        super(label, collapsibleState)
+    }
+
+    protected async handleErrorProneOperation() {
```

Comment:
Rather than force each derived class to have only one operation that could be error handled, could this be opened up to allow arbitrary operations to be error handled? For example, an action could be passed into the function for try/catch wrapping.

Reply:
Good insight!